### ISSUE TITLE
Final MSE calculation in the IterativeClosestPoint filter does not account for the "Maximum correspondence distance" argument

### ISSUE THREAD

**Issue by pjhartzell** [PDAL/PDAL#3471]
If the `max_dist` argument (maximum correspondence distance) is set, the algorithm and the mean square error (MSE) computation inside the main iterative loop only include point pairs with point to point distances less than `max_dist`. This is as expected. Lines 170-197 in IterativeClosestPoint.cpp:
```
        // For every point in the centered, moving PointView, find the nearest
        // neighbor in the centered fixed PointView. Record the indices of each
        // and update the MSE.
        for (PointRef p : *tempMovingTransformed)
        {
            // Find the index of the nearest neighbor, and the square distance
            // between each point.
            PointIdList indices(1);
            std::vector<double> sqr_dists(1);
            kd_fixed.knnSearch(p, 1, &indices, &sqr_dists);


            // In the PCL code, there would've been a check that the square
            // distance did not exceed a threshold value.
            if (m_maxdistArg->set())
            {
                if (sqr_dists[0] > sqr_maxdist)
                    continue;
            }


            // Store the indices of the correspondence and update the MSE.
            moving_idx.push_back(p.pointId());
            fixed_idx.push_back(indices[0]);
            mse += std::sqrt(sqr_dists[0]);
        }


        // Finalize and log the MSE.
        mse /= moving_idx.size();
        log()->get(LogLevel::Debug2) << "MSE: " << mse << std::endl;
```

Once converged and the main iterative loop is exited, the MSE is computed one last time for eventual storage in metadata. In this case, however, the MSE includes all point pairs - the `max_dist` argument value is not used. Lines 281-293:
```
    // Compute the MSE one last time, using the unaltered, fixed PointView and
    // the transformed, moving PointView.
    double mse(0.0);
    KD3Index& kd_fixed_orig = fixed->build3dIndex();
    for (PointRef p : *moving)
    {
        PointIdList indices(1);
        std::vector<double> sqr_dists(1);
        kd_fixed_orig.knnSearch(p, 1, &indices, &sqr_dists);
        mse += std::sqrt(sqr_dists[0]);
    }
    mse /= moving->size();
    log()->get(LogLevel::Debug2) << "MSE: " << mse << std::endl;
```
I think the final MSE value should reflect what actually happened in the iterative loop. A `max_dist` check similar to that inside the iterative loop should be included in the final MSE computation. Currently, if `max_dist` is set, it is possible for the algorithm to converge via the `mse_abs` threshold, yet report an MSE greater than the `mse_abs` threshold in the metadata.

**Comment by pjhartzell** [comment]
I'll submit a PR for this.

**Comment by abellgithub** [comment]
Fixed in #3475
